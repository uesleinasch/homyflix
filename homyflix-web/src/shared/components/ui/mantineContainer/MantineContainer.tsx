import type { JSX, ReactNode } from 'react'
import styles from './style.module.css'
import { Container, ScrollArea, type ContainerProps } from '@mantine/core'

interface CustomContainerProps extends Omit<ContainerProps, 'children'> {
  children: ReactNode
  maxHeight?: string | number
  scrollbarSize?: number
  scrollType?: 'scroll' | 'hover' | 'always'
  enableAutosize?: boolean
}

const MantineContainer = ({ 
  children, 
  maxHeight = 'calc(100vh - 120px)',
  scrollbarSize = 8,
  scrollType = 'scroll',
  enableAutosize = true,
  ...props 
}: CustomContainerProps): JSX.Element => {
  
  const containerContent = enableAutosize ? (
    <ScrollArea.Autosize 
      mah={maxHeight}
      type={scrollType}
      scrollbarSize={scrollbarSize}
      scrollHideDelay={1500}
      style={{ flex: 1, width: '100%' }}
    >
      {children}
    </ScrollArea.Autosize>
  ) : (
    children
  );

  return (
    <Container
      className={styles.container}
      fluid
      miw="100%"
      style={{ 
        minHeight: '100%', 
        height: 'auto',
        display: 'flex',
        flexDirection: 'column',
        flexGrow: 1,
        overflow: enableAutosize ? 'hidden' : 'auto'
      }}
      {...props}
    >
      {containerContent}
    </Container>
  )
}

export default MantineContainer
