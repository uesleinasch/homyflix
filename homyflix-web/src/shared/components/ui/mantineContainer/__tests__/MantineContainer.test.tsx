import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import MantineContainer from '../MantineContainer';

describe('MantineContainer', () => {
  it('Should render children content correctly', () => {
    const { container } = render(
      <MantineProvider>
        <MantineContainer>
          <div>Test content for MantineContainer</div>
        </MantineContainer>
      </MantineProvider>
    );
    
    const content = container.querySelector('div');
    expect(content).toBeInTheDocument();
    expect(content?.innerHTML).toContain('Test content for MantineContainer');
  });

  it('Should match snapshot', () => {
    const { container } = render(
      <MantineProvider>
        <MantineContainer>
          <div>Test content for snapshot</div>
        </MantineContainer>
      </MantineProvider>
    );
    
    expect(container.firstChild).toMatchSnapshot();
  });
}); 