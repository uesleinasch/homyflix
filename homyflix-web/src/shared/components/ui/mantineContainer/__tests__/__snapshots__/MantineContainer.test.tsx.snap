// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MantineContainer Should match snapshot 1`] = `
<style
  data-mantine-styles="classes"
>
  @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
</style>
`;
